### PR TITLE
Todo List

### PR DESCRIPTION
**Pull request by NuffSaid-Bore**
**MERN stack Todo List**
 Application is a simple web app built using the following:
============
BACKEND
============
**Node.js** 
**Express** 

============
FRONTEND
============
**React** 
**Vite**

===========
DATABASE
===========
**MongoDB** 
for storing user data. 

The application allows users to create, read, update, and delete todo tasks. The backend handles API requests, interacts with the MongoDB database to store tasks, while the frontend provides an interactive user interface for managing the todo list. This setup leverages the full power of the MERN stack, providing a fast, scalable, and dynamic user experience.

### CODE REVIEW

**Comment by NuffSaid-Bore** [comment]
@Sinawo, @BAHLEMAJOLA, @zolwethu, @TshepoMagagula This is how my project currently looks like

Screenshot 2024-11-12 130426
Screenshot 2024-11-12 130505
I look forward to your reviews

**Review by zolwethu — CHANGES_REQUESTED** [review]
@NuffSaid-Bore  please move the "incomplete" and "compete" buttons up a notch, so they are in the same card as the "To Do List" header/title

File: Frontend/src/components/NoteModal.jsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+// eslint-disable-next-line react/prop-types
+const NoteModal = ({closeModal}) => {
+
+    const [title, setTitle] = useState('')
+    const [description, setDescription] = useState('')
+
+    const handleSubmit = async (e) => {
+        e.preventDefault();
+        try{
+            const response = await axios.post("http://localhost:5000/add",
```
Comment (BAHLEMAJOLA) [inline Frontend/src/components/NoteModal.jsx]: @NuffSaid-Bore please display a user-friendly error message on the modal if the request fails.
for example; catch (error) {
        setError('Failed to add the task. Please try again .');
        console.log(error);

**Review by BAHLEMAJOLA — CHANGES_REQUESTED** [review]
@NuffSaid-Bore please display a user-friendly error message on the modal if the request fails.
for example; catch (error) {
setError('Failed to add the task. Please try again .');
console.log(error);

**Comment by NuffSaid-Bore** [comment]
> @NuffSaid-Bore please move the "incomplete" and "compete" buttons up a notch, so they are in the same card as the "To Do List" header/title

@zolwethu Thanks for your review, this is the intended design, Please elaborate request for change in the Frontend perspective

**Comment by NuffSaid-Bore** [comment]
> @NuffSaid-Bore please display a user-friendly error message on the modal if the request fails. for example; catch (error) { setError('Failed to add the task. Please try again .'); console.log(error);

@BAHLEMAJOLA I appreciate the review, please find attached screenshot for your requested changes. I already included the error and success validations.
Screenshot 2024-11-13 120147

**Review by TshepoMagagula — COMMENTED** [review]
@NuffSaid-Bore I'm happy with the project, it is responsive and the README is comprehensive, I give you a go ahead to merge the pull request

**Comment by NuffSaid-Bore** [comment]
> @NuffSaid-Bore I'm happy with the project, it is responsive and the README is comprehensive, I give you a go ahead to merge the pull request

@TshepoMagagula The review is much appreciated! awaiting other team members for their go ahead.

**Review by BAHLEMAJOLA — APPROVED** [review]
Great you can now merge 💯

**Comment by zolwethu** [comment]
> > @NuffSaid-Bore please move the "incomplete" and "compete" buttons up a notch, so they are in the same card as the "To Do List" header/title
> 
> @zolwethu Thanks for your review, this is the intended design, Please elaborate request for change in the Frontend perspective

@NuffSaid-Bore Alright, but at least move the Title "To Do List" to the middle of the card

**Comment by NuffSaid-Bore** [comment]
> > > @NuffSaid-Bore please move the "incomplete" and "compete" buttons up a notch, so they are in the same card as the "To Do List" header/title
> > 
> > 
> > @zolwethu Thanks for your review, this is the intended design, Please elaborate request for change in the Frontend perspective
> 
> @NuffSaid-Bore Alright, but at least move the Title "To Do List" to the middle of the card

@zolwethu I have intergrated the requested changes.

**Review by zolwethu — APPROVED** [review]
Title should almost always be centered, looks goodnow

**Pull request merged**